### DISCUSSION TITLE
weird shell behavior in tasks, wildcard not read correctly?

### DISCUSSION THREAD

**Discussion by sg-qwt** [babashka/babashka#1535] [Q&A] [answered]
Again, following babashka book and playing with tasks, noticed a very weird thing:
```
> tree .github
.github
└── workflows
    └── main.yaml
```
```edn
{:tasks
 {clean (shell "rm -rf .github/workflows/*")
  clean2 (shell "rm -rf .github/workflows")}}
```

1. manually running `rm -rf .github/workflows/*` has no issue delete main.yaml
2. running `bb clean` does **not** delete main.yaml, empty output, no error
3. running `bb clean2` does delete workflows folder as whole as expected

Maybe shell has issue with wildcard? 
Related https://github.com/babashka/process/issues/103. An echo maybe helpful to figure out what's going on this case.

## related versions
babashka v1.3.176

Clojure CLI version 1.11.1.1257

java --version
openjdk 11.0.18 2023-01-17
OpenJDK Runtime Environment (build 11.0.18+0-adhoc..source)
OpenJDK 64-Bit Server VM (build 11.0.18+0-adhoc..source, mixed mode)

platform
Linux 6.1.20 NixOS x86_64 GNU/Linux

**Comment by borkdude** [comment, marked as answer]
@sg-qwt I converted your issue to a Q/A discussion over here, I think the Discussions part is more suited for this.

`shell` shells out to an external program, but it does not actually start a bash/zsh or whatever shell. As such it doesn't understand bash syntax.

You can accomplish the above with: 

``` clojure
{:tasks
  {:requires ([babashka.fs :as fs])
   clean (fs/delete-tree ".github/workflows")
```

If you want wildcards etc to work, you have to invoke bash manually like this:

``` clojure
(shell "bash -c 'rm -rf .github/workflows/*'")
```

**Reply by sg-qwt** [reply]
Thanks @borkdude. 
Does it make sense to use what ever $SHELL is setup as a default and fall back to the external shell if $SHELL not setup this case? Most linux system should have some kind of default $SHELL I believe.

Otherwise, this shell behavior is worth documented on the website maybe. Again, thanks for the help!

**Reply by borkdude** [reply]
Bb aims to be a cross-platform scripting tool and I'm afraid `SHELL` is only defined on *nix systems, not on Windows. Using babashka.fs for file-system operations is preferred as this will work everywhere and not only for those people on *nix systems.
Where  / which website should this be clarified? It will be very useful to me to have your feedback on this.

**Reply by sg-qwt** [reply]
A small notice section under https://book.babashka.org/#_shell would be fine by me.

**Reply by borkdude** [reply]
Done!
